File: src/contests/cg/puzzle/kotlin/teds-compiler.kt
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

fun main() {
    val line = readLine()!!.trimEnd()
    var balance = 0
    var best = 0
    for (i in line.indices) {
        if (line[i] == '<') balance++ else balance--
        if (balance < 0) break
        if (balance == 0) best = i + 1
    }
    println(best)
}

File: src/contests/cg/puzzle/fsharp/teds-compiler.fs
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

let line = System.Console.ReadLine().TrimEnd()
let mutable balance = 0
let mutable best = 0
let mutable i = 0
let mutable stop = false
while i < line.Length && not stop do
    if line.[i] = '<' then balance <- balance + 1 else balance <- balance - 1
    if balance < 0 then stop <- true
    else
        if balance = 0 then best <- i + 1
        i <- i + 1
printfn "%d" best
